src/gui/transferlistsortmodel.cpp

```diff
@@ -39,6 +39,44 @@
 
 namespace
 {
+    bool operator<(const QVariant &left, const QVariant &right)
+    {
+        if (left.userType() == QMetaType::UnknownType)
+            return false;
+
+        if (right.userType() == QMetaType::UnknownType)
+            return true;
+
+        switch (left.userType())
```

Comment:
maybe also add an assert that will check `left` and `right` have the same `userType`.